test/integration/job/job_test.go

```diff
@@ -682,6 +682,636 @@ func TestJobPodFailurePolicy(t *testing.T) {
+				},
+			},
+			wantJobConditionType: batchv1.JobFailed,
+		},
+		"pod failure matching pod failure policy rule with FailIndex action": {
+			job: batchv1.Job{
+				Spec: batchv1.JobSpec{
+					Parallelism:          pointer.Int32(2),
+					Completions:          pointer.Int32(2),
+					CompletionMode:       completionModePtr(batchv1.IndexedCompletion),
+					BackoffLimitPerIndex: pointer.Int32(1),
+					Template:             podTemplateSpec,
+					PodFailurePolicy: &batchv1.PodFailurePolicy{
+						Rules: []batchv1.PodFailurePolicyRule{
+							{
+								Action: batchv1.PodFailurePolicyActionFailIndex,
+								OnExitCodes: &batchv1.PodFailurePolicyOnExitCodesRequirement{
+									Operator: batchv1.PodFailurePolicyOnExitCodesOpIn,
+									Values:   []int32{13},
+								},
+							},
+							{
+								Action: batchv1.PodFailurePolicyActionFailIndex,
+								OnPodConditions: []batchv1.PodFailurePolicyOnPodConditionsPattern{
+									{
+										Type:   v1.DisruptionTarget,
+										Status: v1.ConditionTrue,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			podTerminations: []podTerminationWithExpectations{
+				{
+					index: 0,
+					status: v1.PodStatus{
+						Phase: v1.PodFailed,
+						ContainerStatuses: []v1.ContainerStatus{
+							{
+								State: v1.ContainerState{
+									Terminated: &v1.ContainerStateTerminated{
+										ExitCode: 13,
+									},
+								},
+							},
+						},
+					},
+					wantActive:        1,
+					wantFailed:        1,
+					wantActiveIndexes: sets.New(1),
+					wantFailedIndexes: pointer.String("0"),
+				},
+				{
+					index: 1,
+					status: v1.PodStatus{
+						Phase: v1.PodFailed,
+						Conditions: []v1.PodCondition{
+							{
+								Type:   v1.DisruptionTarget,
+								Status: v1.ConditionTrue,
+							},
+						},
+					},
+					wantFailed:        2,
+					wantFailedIndexes: pointer.String("0,1"),
+				},
+			},
+			wantJobConditionType: batchv1.JobFailed,
+		},
+	}
+	for name, test := range testCases {
+		t.Run(name, func(t *testing.T) {
+			resetMetrics()
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobPodFailurePolicy, true)()
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobBackoffLimitPerIndex, true)()
+
+			closeFn, restConfig, clientSet, ns := setup(t, "simple")
+			defer closeFn()
+			ctx, cancel := startJobControllerAndWaitForCaches(restConfig)
+			defer func() {
+				cancel()
+			}()
+			jobObj, err := createJobWithDefaults(ctx, clientSet, ns.Name, &test.job)
+			if err != nil {
+				t.Fatalf("Error %q while creating the job %q", err, jobObj.Name)
+			}
+			validateJobPodsStatus(ctx, t, clientSet, jobObj, podsByStatus{
+				Active: int(*test.job.Spec.Parallelism),
+				Ready:  pointer.Int32(0),
+			})
+			for _, podTermination := range test.podTerminations {
+				op := func(p *v1.Pod) {
+					p.Status = podTermination.status
+				}
```

Comment:
why is this a function?

Reply:
I thought I need something more generic during developments, cleaned up now as not needed.